plugins/analytics-module-new-relic/src/apis/implementations/AnalyticsApi/NewRelicBrowser.ts

```diff
@@ -0,0 +1,113 @@
+  AnalyticsApi,
+  IdentityApi,
+  AnalyticsEvent,
+} from '@backstage/core-plugin-api';
+import { BrowserAgent } from '@newrelic/browser-agent/loaders/browser-agent';
+import type { setAPI } from '@newrelic/browser-agent/loaders/api/api';
+
+type NewRelicAPI = ReturnType<typeof setAPI>;
+
+type NewRelicBrowserOptions = {
+  accountId: string;
+  applicationId: string;
+  licenseKey: string;
+  distributedTracingEnabled: boolean;
+  cookiesEnabled: boolean;
+  useEuEndpoint: boolean;
+};
+
+/**
+ * New Relic Browser API provider for the Backstage Analytics API.
+ * @public
+ */
+export class NewRelicBrowser implements AnalyticsApi {
+  private readonly agent: NewRelicAPI;
+
+  private constructor(
+    options: NewRelicBrowserOptions,
+    identityApi?: IdentityApi,
+  ) {
+    // Configure the New Relic Browser agent
+    const agentOptions = {
+      init: {
+        distributed_tracing: {
+          enabled: options.distributedTracingEnabled,
+        },
+        privacy: {
+          cookies_enabled: options.cookiesEnabled,
+        },
+        ajax: {
+          deny_list: [
+            options.useEuEndpoint ? 'bam.eu01.nr-data.net' : 'bam.nr-data.net',
+          ],
+        },
+      },
+      info: {
+        beacon: options.useEuEndpoint
+          ? 'bam.eu01.nr-data.net'
+          : 'bam.nr-data.net',
+        errorBeacon: options.useEuEndpoint
+          ? 'bam.eu01.nr-data.net'
+          : 'bam.nr-data.net',
+        licenseKey: options.licenseKey,
+        applicationID: options.applicationId,
+        sa: 1,
+      },
+      loader_config: {
+        accountID: options.accountId,
+        trustKey: options.accountId,
+        agentID: options.applicationId,
+        licenseKey: options.licenseKey,
+        applicationID: options.applicationId,
+      },
+    };
+
+    // Initialize the agent
+    this.agent = new BrowserAgent(agentOptions) as unknown as NewRelicAPI;
+
+    if (identityApi) {
+      identityApi.getBackstageIdentity().then(identity => {
+        this.agent.setUserId(identity.userEntityRef);
+      });
+    }
+  }
+
+  static fromConfig(config: Config, options: { identityApi?: IdentityApi }) {
+    const browserOptions: NewRelicBrowserOptions = {
+      accountId: config.getString('app.analytics.nr.accountId'),
+      applicationId: config.getString('app.analytics.nr.applicationId'),
+      licenseKey: config.getString('app.analytics.nr.licenseKey'),
+      distributedTracingEnabled:
+        config.getOptionalBoolean(
+          'app.analytics.nr.distributedTracingEnabled',
+        ) ?? false,
+      cookiesEnabled:
+        config.getOptionalBoolean('app.analytics.nr.cookiesEnabled') ?? false,
+      useEuEndpoint:
+        config.getOptionalBoolean('app.analytics.nr.useEuEndpoint') ?? false,
+    };
+    return new NewRelicBrowser(browserOptions, options.identityApi);
+  }
+
+  captureEvent(event: AnalyticsEvent) {
+    const { action, ...rest } = event;
+    this.agent.addPageAction(action, rest);
+  }
```

Comment:
It's been a minute since I used NR, but I recall that they have a way of differentiating page views from page actions.

My recommendation would be to hook onto events with action "navigate" and send them along as page views, and then let the remaining actions pass as page actions.